security-analysis/security-analysis-api/src/main/java/com/powsybl/security/LimitViolation.java

```diff
@@ -61,38 +66,88 @@ public LimitViolation(String subjectId, LimitViolationType limitType, double lim
         this(subjectId, null, limitType, limit, limitReduction, value);
     }
 
+    /**
+     * The identifier of the network equipment on which the violation occurred.
+     *
+     * @return the identifier of the network equipment on which the violation occurred.
+     */
     public String getSubjectId() {
         return subjectId;
     }
 
+    /**
+     * The name of the network equipment on which the violation occurred.
+     * May be {@code null}.
+     *
+     * @return the name of the network equipment on which the violation occurred.
+     */
+    @Nullable
     public String getSubjectName() {
         return subjectName;
     }
 
+    /**
+     * The type of limit which has been violated.
+     * @return the type of limit which has been violated.
+     */
     public LimitViolationType getLimitType() {
         return limitType;
     }
 
+    /**
+     * The value of the limit which has been violated.
+     * @return the value of the limit which has been violated.
+     */
     public double getLimit() {
         return limit;
     }
 
+    /**
+     * The name of the limit which has been violated. May be {@code null}.
+     * @return the value of the limit which has been violated. May be {@code null}.
+     */
+    @Nullable
     public String getLimitName() {
         return limitName;
     }
 
+    /**
+     * The acceptable duration, in seconds, associated to the current violation value.
+     * Only relevant for current limits.
```

Comment:
I was wondering if we should document this:
if the value is considered meaningless, maybe it should not be documented in the public API so that users do not rely on it ? They should never read and use this value.